src/App.js

```diff
@@ -35,21 +37,76 @@ const styles = theme => ({
 });
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      authContext: {
+        authData: {},
+        isAuthenticated: this.isAuthenticated,
+        setCurrentUser: this.setCurrentUser,
+        setCurrentUserProfile: this.setCurrentUserProfile
+      }
+    };
+  }
+
+  componentDidMount = () => {
+    // Sync auth session stored in the localStorage with Context
+    // TBH, I'm not sure if we even need context
+    this.setCurrentUser(AuthService.getSessionAuthData());
```

Comment:
Instead of calling this here and having a rerender, can we just call `AuthService.getSessionAuthData()` in the constructor?